### PR TITLE
Adding Trivy code scanner

### PR DESCRIPTION
**Pull request by diogormendes**
[Trivy](https://github.com/aquasecurity/trivy) scans for vulnerabilities in container images, file systems, and Git repositories, as well as for configuration issues and detects vulnerabilities of OS packages (Alpine, RHEL, CentOS, etc.) and language-specific packages (Bundler, Composer, npm, yarn, etc.)

The results of the scans are uploaded to [GitHub Code scanning alerts](https://github.com/metabase/metabase/security/code-scanning)

### CODE REVIEW

File: .github/workflows/trivy.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Build an image from Dockerfile
+        run: |
+          docker build -t docker.io/metabase/metabase:${{ github.sha }} .
+
+      - name: Run Trivy vulnerability scanner
```
Comment (paoliniluis) [inline .github/workflows/trivy.yml:30]: why don't we do this at the end of https://github.com/metabase/metabase/blob/master/.github/workflows/uberjar.yml?

**Comment by paoliniluis** [comment]
Just left my comment on the code, I suggest that for saving runners we do that at the end of the container push in the uberjar workflow

File: .github/workflows/trivy.yml (hunk 1 above)
Comment (diogormendes) [reply .github/workflows/trivy.yml:30]: I thought about that, but at the same time, I wanted to make things independently and also not run at every commit.

Maybe we can add

```
concurrency:
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true
```

 on the `uberjar.yml` and that will help on the nodes as well.

**Review by dragonsahead — APPROVED** [review]
LGTM

**Review by rlotun — APPROVED** [review]

**Pull request merged**